pkg/providers/amifamily/ami.go

```diff
@@ -88,15 +88,13 @@ func (p *DefaultProvider) List(ctx context.Context, nodeClass *v1.EC2NodeClass)
 func (p *DefaultProvider) DescribeImageQueries(ctx context.Context, nodeClass *v1.EC2NodeClass) ([]DescribeImageQuery, error) {
 	// Aliases are mutually exclusive, both on the term level and field level within a term.
 	// This is enforced by a CEL validation, we will treat this as an invariant.
-	if term, ok := lo.Find(nodeClass.Spec.AMISelectorTerms, func(term v1.AMISelectorTerm) bool {
-		return term.Alias != ""
-	}); ok {
+	if alias := nodeClass.Alias(); alias != nil {
 		kubernetesVersion, err := p.versionProvider.Get(ctx)
 		if err != nil {
 			return nil, fmt.Errorf("getting kubernetes version, %w", err)
 		}
-		amiFamily := GetAMIFamily(v1.AMIFamilyFromAlias(term.Alias), nil)
-		query, err := amiFamily.DescribeImageQuery(ctx, p.ssmProvider, kubernetesVersion, v1.AMIVersionFromAlias(term.Alias))
+
+		query, err := GetAMIFamily(alias.Family, nil).DescribeImageQuery(ctx, p.ssmProvider, kubernetesVersion, alias.Version)
```

Comment:
musing: I've always hated having to pass `nil` into here -- it's a signal to me that something isn't quite right with the abstraction -- but I figure you already know that -- something to keep an eye out for

Reply:
Agreed, long term this abstraction should probably change, maybe two separate types (one which requires options, one which is independent). I don't want to couple these changes to that refactor though.